### ISSUE TITLE
JWTAuth::refresh generates an already blacklisted token

### ISSUE THREAD

**Issue by akanevsky** [tymondesigns/jwt-auth#602]
I am working on a solution that ensures that there are no concurrent token refresh requests. As a result, I was able to identify and verify a problem, which is that the `jti` of a refreshed token occasionally seems to end up in the blacklist before anyone had a chance to use it in a subsequent request. After spending the entire day on this problem, I wasn't able to determine the root cause. My code follows.

First, I added a temporary function to the core Blacklist class that assisted me in this verification:

vendor/tymon/jwt-auth/src/Blacklist.php (excerpt)
```
    public function hasMy($payload)
    {
        return $this->storage->has($payload->jti);
    }
```

Second, I wrote the following code to check whether a new token is blacklisted before it is even sent to the client. This code is part of my custom middleware.

Middleware\RefreshToken.php (excerpt)
```
$token = JWTAuth::getToken();
$newToken = JWTAuth::refresh($token);
$tok = explode('.', $newToken);
$tok = json_decode(base64_decode($tok[1]));
$bl = JWTAuth::manager()->getBlacklist();
$response->headers->set('Authorization', 'Bearer '.$newToken);
$response->headers->set('AlreadyBlacklisted', $bl->hasMy($tok) ? 'Yes' : 'No');
```

When this code generates a new token, it also sets the AlreadyBlacklisted header to Yes, and any subsequent requests with this new token cause a "blacklisted token" exception.

Please let me know whether there is anything that I might be doing wrong.

Thanks.

**Comment by akanevsky** [comment]
It further seems that the `jti` of the new token is equal to the `jti` of the old token. How can this be?

**Comment by akanevsky** [comment]
I see that the `jti` can be the same when two requests are very close in time. This is because of how `jti` is generated in the current version of the package. I temporarily circumvented the problem by implementing a custom refresh process that forces a unique `jti` in every new token. I see that this has been solved (#364) in the `develop` branch, so I'm closing the issue.
